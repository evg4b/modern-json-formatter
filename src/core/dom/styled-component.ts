import { registerStyle } from '@core/ui/helpers';

export abstract class StyledComponentElement extends HTMLElement {
  protected readonly shadow = this.attachShadow({ mode: 'closed' });
  protected readonly _styles: HTMLStyleElement;

  protected constructor(styles: string) {
    super();
    this._styles = registerStyle(this.shadow, styles);
  }
}
