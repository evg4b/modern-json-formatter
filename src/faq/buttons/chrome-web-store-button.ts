import { CustomElement } from '@core/dom';
import { BaseButtonElement } from './button';
import icon from './chrome-web-store-button-icon.svg';

@CustomElement('chrome-web-store-button')
export class ChromeWebStoreButton extends BaseButtonElement {
  constructor() {
    super(icon);
  }
}
