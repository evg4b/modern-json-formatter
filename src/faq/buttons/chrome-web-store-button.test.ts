import '@testing/svg.mock';
import { describe, expect, it } from '@jest/globals';
import { getShadowRoot } from '@testing/styled-component';
import { ChromeWebStoreButton } from './chrome-web-store-button';

describe('chrome-web-store-button', () => {
  it('should render button link', () => {
    const button = new ChromeWebStoreButton();
    button.attributeChangedCallback('href', '', 'https://google.com');
    button.attributeChangedCallback('title', '', 'Chrome Web Store');
    const root = getShadowRoot(button);
    expect(root.innerHTML).toMatchSnapshot();
  });
});
