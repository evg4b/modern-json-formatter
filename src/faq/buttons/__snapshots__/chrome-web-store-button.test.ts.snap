// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chrome-web-store-button should render button link 1`] = `"<style type="text/css" rel="stylesheet" role="presentation"></style><a target="_blank" href="https://google.com" title="Chrome Web Store"><svg>chrome-web-store-button-icon</svg></a>"`;
